### DISCUSSION TITLE
Is it possible to reference the theme ? (Using vite)

### DISCUSSION THREAD

**Discussion by noook** [windicss/windicss#314] [Q&A]
When I try to import the theme to reference it with Vite I get:

`SyntaxError: import not found: default`

I tried with relative path `../../..` to make sure it's not the alias resolving that's buggy.

Is there any helper to retrieve the loaded theme instead ?

I did a workaround by writing the theme in my constants folder when the options are resolved:

```ts
export default defineConfig({
  plugins: [
    vue(),
    WindiCSS({
      onOptionsResolved(options) {
        writeFileSync(path.resolve(__dirname, 'src', 'constants', 'theme.json'), JSON.stringify(options.config.theme, null, 2));
      },
    }),
  ],
  resolve: {
    alias: {
      '@': `${path.resolve(__dirname, 'src')}`,
    },
  },
  optimizeDeps: {
    exclude: [
      '@windicss/plugin-utils',
    ],
  },
});

```

**Comment by schelmo** [comment]
i made an example here:
https://github.com/schelmo/vite-use-tailwindconfig/tree/windi
in main.js you can see how to get the full config

**Reply by shiftgeist** [reply]
For simplicity:

```js
import Processor from 'windicss'
import windiConfigSource from './windi.config.js'

const windi = new Processor(windiConfigSource)
const windiConfig= processor.allConfig

console.log(windiConfig, windi.interpret('bg-gray-800 text-white').styleSheet.build())
```
